tests/test_verifylib.py

```diff
@@ -1456,6 +1459,38 @@ def test_verify_link_signature_thresholds__S_S_S(self):
     verify_link_signature_thresholds(layout, chain_link_dict)
 
 
+  def test_verify_subkey_thresholds(self):
+    """Subkeys of same main key count only once towards threshold. """
+
+    masterkey = "40e692c3ae03f6b88dff95d0d2c9fe930766998d"
+    subkey1 = "35830aa342b9fea0178876b02b25647ff0ef59fe"
+    subkey2 = "732d722578f71a9ec967a64bfead922c91eb7351"
+
+    link1 = Metablock(signed=Link(name=self.step_name))
+    link1.sign_gpg(subkey1, self.gnupg_home)
+    link2 = Metablock(signed=Link(name=self.step_name))
+    link2.sign_gpg(subkey2, self.gnupg_home)
+
+    chain_link_dict = {
+      self.step_name : {
+        subkey1: link1,
+        subkey2: link2
+      }
+    }
+
+    layout = Layout(
+      steps=[
+          Step(name=self.step_name, pubkeys=[masterkey], threshold=2)
+        ],
+      keys={
+          masterkey: in_toto.gpg.functions.gpg_export_pubkey(
+              masterkey, self.gnupg_home)
+        }
+      )
+    with self.assertRaises(ThresholdVerificationError):
+      verify_link_signature_thresholds(layout, chain_link_dict)
+
+
```

Comment:
it may be overkill, but can we add another one in which we sign with the masterkey and either subkey1 or subkey2?

Reply:
I briefly considered this too, but then I realized that we currently can't sign with a master key, if it has any attached subkeys with signing capabilities, of which gpg uses the most recently added, even if you pass the master key id.  :)

It's possible to forcefully use the masterkey, by appending a `!` to the keyid. However, our keyid schema (`[a-fA-F0-9]`) does not allow that.

Reply:
Ah, fair point. Let's leave this test as-is for now.